module/authentication/src/Application/Security/Provider/RefreshUserProvider.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Authentication\Application\Security\Provider;
+
+use Ergonode\Account\Domain\Repository\UserRepositoryInterface;
+use Ergonode\Authentication\Application\Security\User\User;
+use Ergonode\SharedKernel\Domain\Aggregate\UserId;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class RefreshUserProvider implements UserProviderInterface
```

Comment:
Based on other providers I guess IdUserProvider would be more consistent - the provider does not need to know the context it's being used in.

Reply:
Ok, Thanks for the tip.